Review zebras and add data viz clip (#42)
neuroinformatics-unit/course-animals-in-motion: book/06-movement-zebras.qmd

Reviewer: ```suggestion
For this, we first compute each animal's **unit body vector**. These are a scaled version of the body vectors we just computed, normalised to have unit length.
```
Either remove "`movement` provides a convenience function to do this:" or mention what function it is with a link to [API ref](https://movement.neuroinformatics.dev/latest/api/movement.utils.vector.convert_to_unit.html#movement.utils.vector.convert_to_unit)

The resulting array has `(time, space)` dimensions, which means that we have a single herd vector defined at each timestep.

> The norm of the herd vector will be bounded between 0 and 1, because it is the mean of a set of unit vectors. This is convenient because it already gives us an intuition of how aligned the whole herd is. When the herd vector norm is close to 1, it means that the majority of the unit body vectors are aligned. When its norm is close to 0, it means that the unit body vectors are dispersed. The norm of the herd vector is sometimes called **polarization**.

Reviewer: ```suggestion
The norm of the herd vector will be bounded between 0 and 1, because it is the mean of a set of unit vectors. This is convenient because it already gives us an intuition of how aligned the whole herd is. When the herd vector norm is close to 1, it means that the majority of the unit body vectors are aligned. When its norm is close to 0, it means that the unit body vectors are dispersed. The norm of the herd vector is sometimes called **polarisation**.
```
We might have a mix of BrE "-s-" (e.g. "normalise") and AmE "-z-" in this example.

The plot suggests that the herd alternates between periods of higher and lower polarization.

> To visually check this matches the input data, we can plot polarization in time along with the unit body vectors per individual, the herd vector and the drone video. We did that in the animated plot below. The plot on the top right shows the unit body vectors (in black) and the herd vector (in red), whose norm is the polarization value. The purple line under the herd vector is the equivalent unit vector.

Reviewer: ```suggestion
The plot suggests that the herd alternates between periods of higher and lower polarization.

To confirm that these fluctuations correspond to the herd's actual orientations, we can additionally create an animation that plots each individual's unit body vector and the herd vector for every frame, and present this alongside the polarisation-over-time plot and the drone footage. The resulting video is shown below. The top-left panel shows the polarisation trace, while the top-right panel shows the per-frame visualisation of the herd's orientation: the black arrows are the unit body vectors, the red arrow is the herd vector (whose norm is the polarisation value), and the purple line under the herd vector is the equivalent unit vector.
```

::: {#fig-zebras-polarization}
<video src="https://github.com/neuroinformatics-unit/zebras-stitching/releases/download/zebras-clip/zebras_20250912_174038_w1440_crf23.mp4" autoplay loop muted playsinline width="100%"></video>

> Polarization over time alongside the unit body vectors, the herd vector and the drone video.

Reviewer: ```suggestion
Polarisation over time alongside the unit body vectors, the herd vector and the drone footage.
```

:::

> We can now plot the speed of each individual over time as an array.

Reviewer: ```suggestion
We can now plot the speed of each individual over time.
```

ax.get_images()[0].set_clim(0, 6) # cap values at 6 BL/s
```

> The plot suggests that the individuals are quite coordinated also in their change of speed. We can clearly see the four periods of higher speed, which correspond to the four simulated escape events. The white gaps correspond to the samples we discarded, either because the individual was not detected or because its body length fell outside the range we defined above.

Reviewer: ```suggestion
The plot suggests that the individuals change speed in a coordinated way, with four clear peaks matching the four simulated escape events. The white gaps are the samples we discarded, either because the individual was not detected or because its body length fell outside the range we defined above.
```

## Polarization vs speed

> Let's put it all together and inspect how polarization changes with the speed of the herd.

Reviewer: ```suggestion
Let's now bring everything together and examine how polarisation varies with the herd's speed.
```
